From 561674c2cc145a8c909e2a6024bddfbe043e89b0 Mon Sep 17 00:00:00 2001
From: "guoyin.chen" <guoyin.chen@freescale.com>
Date: Mon, 9 Jul 2012 17:37:20 +0800
Subject: [PATCH 305/635] ENGR00216178 Add native window usage setting for
 camera hal

Make sure the buffer been requested is physical continuous

Signed-off-by: guoyin.chen <guoyin.chen@freescale.com>
---
 mx6/libcamera/CameraHal.cpp | 13 ++++++++++++-
 1 file changed, 12 insertions(+), 1 deletion(-)

diff --git a/mx6/libcamera/CameraHal.cpp b/mx6/libcamera/CameraHal.cpp
index 1b73e6f..50867df 100755
--- a/mx6/libcamera/CameraHal.cpp
+++ b/mx6/libcamera/CameraHal.cpp
@@ -764,12 +764,23 @@ namespace android {
     {
         CAMERA_LOG_FUNC;
 
+        status_t err;
         //Mutex::Autolock lock(mLock);
         if (mNativeWindow == NULL){
             CAMERA_LOG_ERR("the native window is null!");
             return NO_ERROR;//BAD_VALUE;
         }
-        status_t err = mNativeWindow->set_buffers_geometry(mNativeWindow,
+
+        //Make sure the buffer be phiscal continuous
+        err = mNativeWindow->set_usage(mNativeWindow, 
+                GRALLOC_USAGE_SW_READ_NEVER | GRALLOC_USAGE_HW_TEXTURE);
+        if(err != 0){
+            CAMERA_LOG_ERR("native_window_set_usage failed:%s(%d)",
+                    strerror(-err), -err);
+            return err;
+        }
+
+        err = mNativeWindow->set_buffers_geometry(mNativeWindow,
                 mCaptureDeviceCfg.width, mCaptureDeviceCfg.height,
                 HAL_PIXEL_FORMAT_YCbCr_420_SP);//mCaptureDeviceCfg.fmt);
         if(err != 0){
-- 
1.8.0

